### ISSUE TITLE
Backprop through `LayerIntegratedGradients` attributions

### ISSUE THREAD

**Issue by aarzchan** [pytorch/captum#762]
Hi! I'm interested in regularizing the model via some attribution-based loss, as described in a [previous post](https://github.com/pytorch/captum/issues/761#issue-996495482).

To do so, I would like to be able to regularize the model using attributions for arbitrary model layers, not just the input layer. If I create `IntegratedGradients` attributions (w.r.t. the input), the attributions have a `grad_fn` and can be backpropagated through. However, if I create `LayerIntegratedGradients` attributions (w.r.t. some layer), the attributions don't have a `grad_fn`. Do you have any tips for how to backpropagate through `LayerIntegratedGradients` attributions?

Thanks for your help!

**Comment by vivekmig** [comment]
Hi @aarzchan , in order to obtain higher-order derivatives / gradients of these attributions, the gradients need to be computed with create_graph=True when calling torch.autograd.grad. There is some discussion of this for other methods in #652 .

For LayerIntegratedGradients in particular, enabling this does require a small change to the method code, so for now, it would be best to make the change and install Captum from source. Particularly, this [line](https://github.com/pytorch/captum/blob/master/captum/attr/_core/layer/layer_integrated_gradients.py#L478) needs to be modified to be:
```
grads = torch.autograd.grad(torch.unbind(output), inputs, create_graph=True)
```

Hope this helps!

**Comment by aarzchan** [comment]
Hi @vivekmig, thanks for the pointers! I'll let you know if I have other questions later.
